control_plane/src/compute.rs

```diff
@@ -334,14 +334,24 @@ impl PostgresNode {
         if let Some(lsn) = self.lsn {
             conf.append("recovery_target_lsn", &lsn.to_string());
         }
+
         conf.append_line("");
+        // Configure backpressure
+        // - Replication write lag depends on speed of inserting data in storage (in inmem-layer) by walreceiver.
+        //   This lag determines latency of get_page_at_lsn. Speed of applying WAL is about 10Mb/sec,
+        //   So to avoid expiration of 1 minute timeout, this lag should not be larger than 600Mb.
+        //   Actually latency should be much smaller (better if < 1sec). But we assume that recently updated
+        //   pages are not requested from pageserver.
+        // - Replication flush lag depends on speed of persisting data by checkpointer (creation of delta/image layers)
+        //   and advancing disk_consistent_lsn. Safekeepers are able to remove/archive WAL only beyond disk_consistent_lsn.
+        //   Too large lag can cause long recovery time (in case of pageserver crash) and disk space overflow at safekeepers.
+        // - Replication apply lag depends on speed of uploading changes to S3 by uploader thread.
+        //   To be able to restore database in case of pageserver node crash, safekeeper should not remove WAL beyond this point.
+        //   Too large lag can cause space exhaustion at S3 and safekeepers (if them are not able to upload WAL to S3).
+        conf.append("max_replication_write_lag", "500MB");
+        conf.append("max_replication_flush_lag", "10GB");
```

Comment:
Do we disable `max_replication_apply_lag` backpressure for now?

Reply:
It may be needed only for S3 integration while it is not properly tested.
It is hard to specify some meaningful value here (because S3 speed, size limits and our requirements to long life storage are still unclear) and choosing some random number seems to be not good idea.

Reply:
Fair enough